billboardd/tklockinterface.h

```diff
@@ -0,0 +1,62 @@
+
+/**
+ * Billboard - Low Power Mode Standby Screen for the N9
+ * Webpage: http://thp.io/2012/billboard/
+ * Copyright (C) 2012, 2013, 2014 Thomas Perl <thp.io/about>
+ * Copyright 2015 Elliot Wolk <elliot.wolk@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+#ifndef TKLOCKINTERFACE_H
+#define TKLOCKINTERFACE_H
+
+#include <QObject>
+
+class TklockInterface : public QObject
+{
+    Q_OBJECT
+
+    public:
+        TklockInterface();
+        ~TklockInterface();
+
+        bool isLocked() {
```

Comment:
It's not strictly necessary any more now that we only listen to the changed signal, but I guess it doesn't hurt to have it in (it will report a wrong value if Billboard is started in locked state, e.g. you switch on the phone, lock it before Billboard is started, and then `isLocked()` will always be `false` (because `m_lock_mode` will be `""`) until the first D-Bus signal comes in.

So I'd be for removing it until we actually need it (and then we might need to do a query on start up).